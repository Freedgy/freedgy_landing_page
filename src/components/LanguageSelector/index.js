import React from 'react';

import { navigate } from 'gatsby';

import ReactFlagsSelect from 'react-flags-select';


function LanguageSelector({ classes, lang, location, className }) {
    const [selected, setSelected] = React.useState(lang == "en" ? 'GB' : lang.toUpperCase());

    function switchLanguage(code) {
        if (code == 'GB') {
            navigate(`/${location.hash}`)
        } else {
            navigate(`/${code.toLowerCase()}${location.hash}`)
        }
    }

    return (
        <ReactFlagsSelect
            selected={selected}
            countries={["GB", "FR", "ES"]}
            showSelectedLabel={false}
            showOptionLabel={false}
            onSelect={code => switchLanguage(code)}
        />
    );
}

export default LanguageSelector;
