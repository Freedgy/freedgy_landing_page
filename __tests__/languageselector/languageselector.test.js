import React from "react"
import renderer from "react-test-renderer"

import LanguageSelector from "../../src/components/LanguageSelector"

import jsdom from 'jsdom';
const {JSDOM} = jsdom;
const {document} = (new JSDOM('<!doctype html><html><body></body></html>')).window;
global.document = document;
global.window = document.defaultView;
global.Image = window.Image;

describe("LanguageSelector", () => {
    it("renders correctly", async () => {
        const datas = require("../../static/pages/pages.en.json");

        const tree = renderer
            .create(<LanguageSelector lang="en" />)
            .toJSON()
        expect(tree).toMatchSnapshot()
    })
})