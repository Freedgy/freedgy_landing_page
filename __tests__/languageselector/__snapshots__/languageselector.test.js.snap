// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LanguageSelector renders correctly 1`] = `
<div
  className="ReactFlagsSelect-module_flagsSelect__2pfa2"
  data-testid="rfs"
>
  <button
    aria-expanded={false}
    aria-haspopup="listbox"
    aria-labelledby="rfs-btn"
    className="ReactFlagsSelect-module_selectBtn__19wW7"
    data-testid="rfs-btn"
    disabled={false}
    id="rfs-btn"
    onClick={[Function]}
    onKeyUp={[Function]}
    style={
      Object {
        "fontSize": 16,
      }
    }
    type="button"
  >
    <span
      className="ReactFlagsSelect-module_selectValue__152eS"
    >
      Select a country
    </span>
  </button>
</div>
`;
